wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -92,13 +94,23 @@ gzip set app env sendResponse = app env $ \res ->
         ResponseFile{} | gzipFiles set == GzipIgnore -> sendResponse res
         _ -> if "gzip" `elem` enc && not isMSIE6 && not (isEncoded res) && (bigEnough res)
                 then
-                    case (res, gzipFiles set) of
-                        (ResponseFile s hs file Nothing, GzipCacheFolder cache) ->
-                            case lookup hContentType hs of
-                                Just m
-                                    | gzipCheckMime set m -> compressFile s hs file cache sendResponse
-                                _ -> sendResponse res
-                        _ -> compressE set res sendResponse
+                    let runAction x = case x of
+                            (_ , GzipPreCompressed (GzipPreCompressed _)) -> sendResponse res
```

Comment:
This isn't what I would have expected. I'd have expected another clause after the `ResponseFile, GzipPreCompressed` clause that looks like:

```haskell
(res, GzipPreCompressed nextAction) -> runAction (res, nextAction)
```